### PR TITLE
Set api

### PR DESCRIPTION
**Pull request by mustafaiman**
Introduces Set API.
Changes MapInterface to IMap so it conforms Hazelcast naming standard.
Introduces DistributedObject.
Introduces ICollection.

### CODE REVIEW

**Comment by asimarslan** [comment]
Some of the parameter type do not match the parent interface like collection parameters, can you review them and make sure the types are consistent.

**Comment by mustafaiman** [comment]
@asimarslan I reviewed type parameter and fixed collection parameters. Let me know if there is still something I am missing.

**Comment by bilalyasar** [comment]
:+1:

File: src/ICollection.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    add(item : E) : Promise<boolean>;
+
+    /**
+     * Ensures that this collection all the items in given collection.
+     * This collection contains all elements of collection and its previous items at the end.
+     * @param collection
+     * @throws {Error} if collection or one of its elements is null or undefined.
+     * @return true if this collection changed, false otherwise.
+     */
+    addAll(collection : ICollection<E>) : Promise<boolean>;
```
Comment (asimarslan) [inline src/ICollection.ts]: All ICollection parameters in this interface are ordinary, some language builtin collection, what ever exist in TypeScript but not ICollection.

File: src/ISet.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    add(entry : E) : Promise<boolean>;
+
+    /**
+     * Adds the elements contained in collection to this set if not already present.
+     * Set contains all elements of collection and its previous elements at the end.
+     * @param collection
+     * @throws {Error} if collection or one of its elements is null or undefined.
+     * @return true if this set changed, false otherwise.
+     */
+    addAll(collection : ICollection<E>) : Promise<boolean>;
```
Comment (asimarslan) [inline src/ISet.ts]: same here for ICollection, see comment in ICollection

**Pull request merged**